### PR TITLE
Add Build Number and Version suffix

### PR DESCRIPTION
**Pull request by LefterisJP**
- CI Builds should provide a build number and optionally a version suffix
  so that a particular build of a version can be easily identified.
- For user builds these are totally optional and have sensible defaults.
- Adding code to include them in BuildInfo.h

### CODE REVIEW

File: cmake/EthOptions.cmake [hunk 1]
```diff
@@ -43,6 +43,21 @@ macro(configure_project)
 		add_definitions(-DETH_JSCONSOLE)
 	endif()
 
+	# CI Builds should provide (for user builds this is totally optional)
+	# -DBUILD_NUMBER - A number to identify the current build with. Becomes TWEAK component of project version.
+	# -DVERSION_SUFFIX - A string to append to the end of the version string where applicable.
+	if (NOT DEFINED BUILD_NUMBER)
+		# default is very big so that local build is always considered greater
+		# and can easily replace CI build for for all platforms if needed
+		set(BUILD_NUMBER 9999999)
```
Comment (arkpar) [inline cmake/EthOptions.cmake]: Windows only supports up to 65535. Not sure about OS X

Comment (chfast) [inline cmake/EthOptions.cmake]: It is too big. Windows allows only 4 digits.

Comment (LefterisJP) [inline cmake/EthOptions.cmake]: So only 2 bytes per version component. Still should be enough. Let me change it.

File: cmake/EthOptions.cmake [hunk 2]
```diff
@@ -43,6 +43,21 @@ macro(configure_project)
...
+	endif()
+	# Suffix like "-rc1" e.t.c. to append to versions wherever needed.
+	if (NOT DEFINED VERSION_SUFFIX)
+		set(VERSION_SUFFIX "")
+	endif()
+	set (PROJECT_VERSION_TWEAK ${BUILD_NUMBER})
+	set (PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.${PROJECT_VERSION_TWEAK}")
```
Comment (chfast) [inline cmake/EthOptions.cmake]: Is it needed already?

**Pull request merged**